google/cloud/bigquery/magics/magics.py

```diff
@@ -89,14 +89,23 @@
 import sys
 import time
 import warnings
+
+import threading
 from concurrent import futures
 
 try:
     import IPython  # type: ignore
-    from IPython import display  # type: ignore
+    from IPython.display import display, HTML, clear_output  # type: ignore
     from IPython.core import magic_arguments  # type: ignore
 except ImportError:  # pragma: NO COVER
     raise ImportError("This module can only be loaded in IPython.")
+try:
+    import ipywidgets as widgets  # type: ignore
+except ImportError:  # pragma: NO COVER
+    raise ImportError(
+        "This module can only be loaded in IPython with ipywidgets installed."
+    )
```

Comment:
with this, we are effectively saying that the BQ magic can only be used with ipywidgets installed. there could be enterprise users for whom existing notebooks will break because they don't have ipywidgets installed, and they don't have permissions to install it (say they are running in hosted notebooks). even if such users don't care about the ability to run async long running cells, they won't be able to use their notebooks. i think we should try import only if args.send_long_job is true.